### PR TITLE
[EventHubs] Updated blob checkpointing API

### PR DESCRIPTION
**Pull request by annatisch**
Should not be merged until renaming PR #8717 has been merged.
Resolves issues: #8650, and #8535 (also contributes to #7808)

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/eventhub/azure-eventhubs-checkpointstoreblob-aio/azure/eventhub/extensions/checkpointstoreblobaio/_blobstoragepmaio.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    :param container_name:
+        The name of the container for the blobs.
+    :type container_name: str
+    :param credential:
+        The credentials with which to authenticate. This is optional if the
+        account URL already has a SAS token. The value can be a SAS token string, an account
+        shared access key, or an instance of a TokenCredentials class from azure.identity.
+        If the URL already has a SAS token, specifying an explicit credential will take priority.
+    """
+    def __init__(self, blob_account_url, container_name, credential=None, **kwargs):
```
Comment (YijunXieMS) [inline sdk/eventhub/azure-eventhubs-checkpointstoreblob-aio/azure/eventhub/extensions/checkpointstoreblobaio/_blobstoragepmaio.py]: Just to confirm, does the **kwargs have anything that needs to reference Storage Blob library?
For this async code, after argument "credential", the **kwargs should be changed to *, arg1=..., arg2=...?

Comment (annatisch) [reply sdk/eventhub/azure-eventhubs-checkpointstoreblob-aio/azure/eventhub/extensions/checkpointstoreblobaio/_blobstoragepmaio.py]: Yup I can use the Py3 "*" syntax here.
Although I'm rather hesistant to include _all_ the possible options for Storage... I'll see if I can filter down a relevant list.

Comment (annatisch) [reply sdk/eventhub/azure-eventhubs-checkpointstoreblob-aio/azure/eventhub/extensions/checkpointstoreblobaio/_blobstoragepmaio.py]: I've used the syntax for the `credential` parameter, however looking at the docs I don't think there's any further keyword args we should expose here - they're largely related to blob upload chunking which isn't relevant.

File: sdk/eventhub/azure-eventhubs-checkpointstoreblob-aio/HISTORY.md [hunk 2]
```diff
@@ -1,5 +1,15 @@
 # Release History
 
+## 2019-12-04 1.0.0b6
+
+**Breaking changes**
+
+- Renamed `BlobPartitionManager` to `BlobCheckpointStore`.
+- Constructor of `BlobCheckpointStore` has been updated to take the storage container details directly rather than an instance of `ContainerClient`.
+- A `from_connection_string` constructor has been added for Blob Storage connection strings.
+- Module `blobstoragepmaio` is now internal, import directly from `azure.eventhub.extensions.checkpointstoreblobaio`.
```
Comment (bryevdv) [inline sdk/eventhub/azure-eventhubs-checkpointstoreblob-aio/HISTORY.md]: Should users be importing it, if it is internal?

**Comment by bryevdv** [comment]
LGTM tho obviously there is the one failed test to figure out. Strangely it looks like all the tests in the job actually passed and then `Fatal Python error: Segmentation fault` in some step *after* the tests?

File: sdk/eventhub/azure-eventhubs-checkpointstoreblob-aio/HISTORY.md (hunk 2 above)
Comment (annatisch) [reply sdk/eventhub/azure-eventhubs-checkpointstoreblob-aio/HISTORY.md]: That's poor wording on my part - I meant to state that if users were importing anything from that module, they should now import directly from checkpointstoreblob instead.....
I will revise.

File: sdk/eventhub/azure-eventhubs-checkpointstoreblob-aio/README.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ This Checkpoint Store package works as a plug-in package to `EventHubConsumerCli
...
 
-[Source code](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/eventhub/azure-eventhubs-checkpointstoreblob-aio) | [Package (PyPi)](https://pypi.org/project/azure-eventhub-checkpointstoreblob-aio/) | [API reference documentation](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-eventhub/5.0.0b5/azure.eventhub.aio.html#azure.eventhub.aio.PartitionManager) | [Azure Eventhubs documentation](https://docs.microsoft.com/en-us/azure/event-hubs/) | [Azure Storage documentation](https://docs.microsoft.com/en-us/azure/storage/)
+[Source code](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/eventhub/azure-eventhubs-checkpointstoreblob-aio) | [Package (PyPi)](https://pypi.org/project/azure-eventhub-checkpointstoreblob-aio/) | [API reference documentation](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-eventhub/5.0.0b5/azure.eventhub.aio.html#azure.eventhub.aio.CheckpointStore) | [Azure Eventhubs documentation](https://docs.microsoft.com/en-us/azure/event-hubs/) | [Azure Storage documentation](https://docs.microsoft.com/en-us/azure/storage/)
```
Comment (yunhaoling) [inline sdk/eventhub/azure-eventhubs-checkpointstoreblob-aio/README.md]: version being 5.0.0b6?

same for the following version occurrences

Comment (annatisch) [reply sdk/eventhub/azure-eventhubs-checkpointstoreblob-aio/README.md]: Updated

File: sdk/eventhub/azure-eventhubs-checkpointstoreblob-aio/HISTORY.md (hunk 2 above)
Comment (annatisch) [reply sdk/eventhub/azure-eventhubs-checkpointstoreblob-aio/HISTORY.md]: Updated

**Pull request merged**